systems/trajectory_optimization/hybrid_dircon.h

```diff
@@ -134,7 +137,27 @@ class HybridDircon
 
   using drake::systems::trajectory_optimization::MultipleShooting::N;
   using drake::systems::trajectory_optimization::MultipleShooting::
-        SubstitutePlaceholderVariables;
+      SubstitutePlaceholderVariables;
+
+  void ScaleTimeVariables(double scale);
+  void ScaleStateVariables(double scale, int idx_start, int idx_end);
```

Comment:
I can't say I'm a fan of this API. Two reasons:

1. The Drake API is `setScaling(variable, scale)`, which I think makes sense. Here, you've reversed that ordering.
2. "start, end" is a very old-fashioned approach. I suppose it's OK, but I think the more basic `ScaleXXVariable(idx, scale)` should also be included (for a single variable) as an option.

Reply:
To scale a variable, we could use `SetVariableScaling` from Drake. I am curious what you think is a less old-fashioned approach. Maybe ScaleXXVariable(idx_list, scale)?

Reply:
I see the functions you've created here a syntactic sugar on top of Drake's `SetVariableScaling` (ways to auto-extract all the state/time/input/etc. variables).

I think what you mentioned `ScaleXXVariable(idx, scale)` makes sense, yes (mentioned in my earlier comment). If you think it's important to be able to set ranges of variables in one call, then I suppose `start,end` is OK. The other alternative would be something like `ScaleXXVariables(vector<int> indices, scale)`, but that's not a very easy API.

Reply:
I changed the API to `ScaleXXVariable(idx_list, scale)`. I think it's not hard to use